homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,115 @@
+
+SENSOR_TYPES = {
+    'Temperature': [TEMP_CELCIUS],
+    'Humidity': ['%'],
+    'CO2': ['ppm'],
+    'Noise': ['dB'],
+    'Pressure': ['mb']
+}
+
+sensor_url = "https://api.netatmo.com//api/getstationsdata?access_token="
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'NETAMTO Sensor'
+DEFAULT_METHOD = 'GET'
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Gets a list of the available stations and sets up the sensors. """
+
+    client_id = config.get('client_id', None)
+    client_secret = config.get('client_secret', None)
+    username = config.get('username', None)
+    password = config.get('password', None)
+    dev = list()
+    request = requests.post("https://api.netatmo.net/oauth2/token",
+                            data={'grant_type': 'password',
+                                  'client_id': client_id,
+                                  'client_secret': client_secret,
+                                  'username': username,
+                                  'password': password})
+    access_token = request.json()["access_token"]
+    request = requests.get(sensor_url + access_token).json()
+    for device in request["body"]["devices"]:
+        for data_type in device["data_type"]:
+            dev.append(NetatmoSensor(hass, client_id, client_secret,
+                                     username, password, data_type,
+                                     device["module_name"]))
+        for module in device["modules"]:
+            for data_type in module["data_type"]:
+                dev.append(NetatmoSensor(hass, client_id, client_secret,
+                                         username, password, data_type,
+                                         module["module_name"]))
+    add_devices(dev)
+
+
+# pylint: disable=too-many-arguments
+class NetatmoSensor(Entity):
+    """ Implements a netatmo sensor. """
+
+    def __init__(self, hass, client_id, client_secret, username,
+                 password, data_type, module_name):
+        self._hass = hass
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._username = username
+        self._password = password
+        self._state = STATE_UNKNOWN
+        self._data_type = data_type
+        self._name = module_name
+        self._unit_of_measurement = SENSOR_TYPES[data_type][0]
+        self.update()
+
+    @property
+    def name(self):
+        """ The name of the sensor. """
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit the value is expressed in. """
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
You can find an explanation and an implementation about the the throttle in #128.